pkg/note/note.go

```diff
@@ -150,3 +168,43 @@ func NewNoteSigner(ctx context.Context, origin string, signer signature.Signer)
 		sign: sign,
 	}, nil
 }
+
+func NewNoteVerifier(ctx context.Context, origin string, verifier signature.Verifier) (note.Verifier, error) {
+	if !isValidName(origin) {
+		return nil, fmt.Errorf("invalid name %s", origin)
+	}
+
+	pubKey, err := verifier.PublicKey()
+	if err != nil {
+		return nil, fmt.Errorf("getting public key: %w", err)
+	}
+
+	var keyID uint32
```

Comment:
This can be extracted into a helper function since it's shared by `NewNoteSigner`